lib/msf/base/sessions/scriptable.rb

```diff
@@ -164,13 +164,17 @@ def execute_script(script_name, *args)
     else
       full_path = self.class.find_script_path(script_name)
 
-      # No path found?  Weak.
       if full_path.nil?
         print_error("The specified script could not be found: #{script_name}")
-        return true
+        return
+      end
+
+      begin
+        execute_file(full_path, args)
+        framework.events.on_session_script_run(self, full_path)
```

Comment:
note a subtle change in behavior here: if the session script fails to run, it doesn't fire the event anymore.